core/db_models/EEM_Datetime.model.php

```diff
@@ -150,40 +154,97 @@ protected function __construct($timezone)
      * @access public
      * @return EE_Datetime[] array on success, FALSE on fail
      * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws ReflectionException
+     * @throws InvalidInterfaceException
      */
     public function create_new_blank_datetime()
     {
         // makes sure timezone is always set.
         $timezone_string = $this->get_timezone();
+        $start_date = apply_filters(
```

Comment:
Maybe a comment here too would be good, explaining the type of the value being filtered.